processor/src/operations/sys_ops.rs

```diff
@@ -92,6 +92,19 @@ where
 
         Ok(())
     }
+
+    // CLOCK CYCLE
+    // --------------------------------------------------------------------------------------------
+
+    /// Pushes the current clock cycle onto the stack. The clock cycle is the number of times the
+    /// `execute_op` method has been called on the VM. The clock cycle is incremented after the
+    /// execution of each operation.
```

Comment:
I wouldn't tie this to the `execute_op` method. Maybe replace this with something like this?:
```
Pushes the current value of the clock cycle counter onto the stack. The clock cycle starts
at 0 and is incremented with every operation executed by the VM, including control flow
operations such as GRUOP, END etc.
```